index.d.ts

```diff
@@ -511,3 +511,27 @@ Menu.setApplicationMenu(menu);
 ```
 */
 export function appMenu(menuItems?: readonly MenuItemConstructorOptions[]): MenuItemConstructorOptions;
+
+/**
+Open the System Preferences on macOS.
+
+This method does nothing on other systems.
+
+Optionally provide a pane and section. A list of available options can be found [here](https://macosxautomation.com/system-prefs-links.html).
+
+@example
+```
+import {openSystemPreferences} from 'electron-util';
+
+openSystemPreferences();
+
+// or
+
+openSystemPreferences('security', 'Firewall');
+```
+
+@param pane - The pane to open.
+@param section - The section within that pane.
+@returns A Promise that resolves when the preferences window is opened.
+*/
+export const openSystemPreferences: (pane?: string, section?: string) => Promise<void>;
```

Comment:
It would be useful to add custom types for pane and section which would contain a union of the possible combinations. That way the user gets autocomplete. The `section` type would depend on the `pane` type.

Reply:
I can pre-define it, but I know that there's some that are not listed in that list.

For example Catalina introduced `...security?Privacy_ScreenCapture` which is not in that list. That list is also missing `Privacy_Microphone`, which I think was Mojave or El Capitan, but in general it will depend on the user's OS version.

I can add that list and then still allow a generic string?

Reply:
I considered adding these and also allow a generic string, but then people have no incentive to submit missing panes/sections.

Reply:
So I think we should hardcode these and get people to submit anything missing we didn’t think about.

Reply:
Do you mean hardcode the list and let people submit issues to Apple? (or whoever owns the list) or also add the ones that I've found, and then have people submit requests here?

Reply:
Reading on the site https://macosxautomation.com/ it says it's not owned by apple, and it looks like he stopped updating after Yosemite. Maybe there's an official list somewhere, that we can reference and use the build the typings? I'll look again, but I couldn't find anything other than this

Reply:
No, we hard-code the list here, and they submit a PR here if new panes/sections are discovered.

Reply:
> Reading on the site macosxautomation.com it says it's not owned by apple, and it looks like he stopped updating after Yosemite.

That's another good reason for us to maintain a list of it ourselves.

Reply:
Should be good 👍